responsibleai/responsibleai/rai_insights/rai_base_insights.py

```diff
@@ -0,0 +1,317 @@
+# Copyright (c) Microsoft Corporation
+# Licensed under the MIT License.
+
+"""Defines the RAIBaseInsights class."""
+
+import json
+import pickle
+import warnings
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import Any, Optional
+
+import pandas as pd
+
+from responsibleai._internal.constants import Metadata
+
+_DATA = 'data'
+_DTYPES = 'dtypes'
+_TRAIN = 'train'
+_TEST = 'test'
+_MODEL = Metadata.MODEL
+_MODEL_PKL = _MODEL + '.pkl'
+_SERIALIZER = 'serializer'
+_MANAGERS = 'managers'
+_JSON_EXTENSION = '.json'
+
+
+class RAIBaseInsights(ABC):
+    """Defines the base class RAIBaseInsights for the top-level API.
+
+    This class is abstract and should not be instantiated.
+    """
+
+    def __init__(self, model: Any, train: pd.DataFrame, test: pd.DataFrame,
+                 target_column: str, task_type: str,
+                 serializer: Optional[Any] = None):
+        """Creates an RAIBaseInsights object.
+
+        :param model: The model to compute RAI insights for.
+            A model that implements sklearn.predict or sklearn.predict_proba
+            or function that accepts a 2d ndarray.
+        :type model: object
+        :param train: The training dataset including the label column.
+        :type train: pandas.DataFrame
+        :param test: The test dataset including the label column.
+        :type test: pandas.DataFrame
+        :param target_column: The name of the label column.
+        :type target_column: str
+        :param task_type: The task to run.
+        :type task_type: str
+        :param classes: The class labels in the training dataset
+        :type classes: numpy.ndarray
+        :param serializer: Picklable custom serializer with save and load
+            methods for custom model serialization.
+            The save method writes the model to file given a parent directory.
+            The load method returns the deserialized model from the same
+            parent directory.
+        :type serializer: object
+        """
+        self.model = model
+        self.train = train
+        self.test = test
+        self.target_column = target_column
+        self.task_type = task_type
+        self._serializer = serializer
+        self._initialize_managers()
+
+    @abstractmethod
+    def _initialize_managers(self):
+        """Initializes the managers.
+
+        This method is abstract and should not be called.
+        """
+        pass
```

Comment:
Am I right in thinking that the `@abstractmethod` decorator will override the `pass` and throw an error if something tries to call this?